src/tools/clippy/clippy_lints/src/almost_complete_letter_range.rs

```diff
@@ -73,12 +73,21 @@ impl EarlyLintPass for AlmostCompleteLetterRange {
 }
 
 fn check_range(cx: &EarlyContext<'_>, span: Span, start: &Expr, end: &Expr, sugg: Option<(Span, &str)>) {
-    if let ExprKind::Lit(start_lit) = &start.peel_parens().kind
-        && let ExprKind::Lit(end_lit) = &end.peel_parens().kind
+    if let ExprKind::Lit(start_token_lit) = start.peel_parens().kind
+        && let ExprKind::Lit(end_token_lit) = end.peel_parens().kind
         && matches!(
-            (&start_lit.kind, &end_lit.kind),
-            (LitKind::Byte(b'a') | LitKind::Char('a'), LitKind::Byte(b'z') | LitKind::Char('z'))
-            | (LitKind::Byte(b'A') | LitKind::Char('A'), LitKind::Byte(b'Z') | LitKind::Char('Z'))
+            (
+                LitKind::from_token_lit(start_token_lit),
+                LitKind::from_token_lit(end_token_lit),
```

Comment:
I tried it, it's a pain. Because the token lit uses symbols, the pattern matching is really clumsy unless I introduced symbols for `A`, `Z`, `a`, and `z`. And then that changes the order of error messages in some tests due to some kind of symbol ordering situation, because those four letters are now ordered as symbols before other single letter symbols. It's all a big hassle and not worth it for something that's not an important part of this PR.